python/solver.py

```diff
@@ -1040,7 +1040,7 @@ def _rootfun(k):
             ks.append(mp.find_root_deriv(rootfun(b), tol, kmag_min, kmag_max, k0s[b - band_min]))
 
         if band_funcs:
-            for b, k in zip(range(band_max, band_max - nb, -1), reversed(ks)):
+            for b, k in zip(range(1, band_max +1), reversed(ks)):
```

Comment:
ks shouldn't be reversed now?

Reply:
No — I think that was a bug before. The Ks are generated in reverse order to begin with.